src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionSimplifyBooleanReturnTest.java

```diff
@@ -0,0 +1,87 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import static com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanReturnCheck.MSG_KEY;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanReturnCheck;
+
+public class XpathRegressionSimplifyBooleanReturnTest extends AbstractXpathTestSupport {
+
+    private static final Class<SimplifyBooleanReturnCheck> CLASS = SimplifyBooleanReturnCheck.class;
+    private final String checkName = CLASS.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testIfBooleanEqualsBoolean() throws Exception {
+        final File fileToProcess = new File(
+            getPath(
+                "SuppressionXpathRegressionSimplifyBooleanReturnIfBooleanEqualsBoolean.java"));
+
+        final DefaultConfiguration moduleConfig = createModuleConfig(CLASS);
+
+        final String[] expectedViolation = {
+            "6:9: " + getCheckMessage(CLASS, MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
+            "/COMPILATION_UNIT/CLASS_DEF[./IDENT[@text="
+                + "'SuppressionXpathRegressionSimplifyBooleanReturnIfBooleanEqualsBoolean']]"
+                + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='toTest']]/SLIST/LITERAL_IF"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+            expectedXpathQueries);
+    }
+
+    @Test
+    public void testIfBooleanReturnBoolean() throws Exception {
+        final File fileToProcess = new File(
+            getPath(
+                "SuppressionXpathRegressionSimplifyBooleanReturnIfBooleanReturnBoolean.java"
+            ));
+
+        final DefaultConfiguration moduleConfig = createModuleConfig(CLASS);
+
+        final String[] expectedViolation = {
+            "6:9: " + getCheckMessage(CLASS, MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
+            "/COMPILATION_UNIT/CLASS_DEF[./IDENT[@text="
+                + "'SuppressionXpathRegressionSimplifyBooleanReturnIfBooleanReturnBoolean']]"
+                + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='toTest']]/SLIST/LITERAL_IF"
```

Comment:
Xpath is almost identical with first test.
Let's put IF of this test input file to lambda, to test that AST path  is different  and it works.

Reply:
Done!